utils/transactions.ts

```diff
@@ -8,3 +8,55 @@ export const getGasEstimateForTransaction = async (txArgs: any[], method: Functi
 		throw error;
 	}
 };
+
+type TxOptions = {
+	showErrorNotification?: Function;
+	showProgressNotification?: Function;
+	showSuccessNotification?: Function;
+};
+
+/**
+ * Perform a contract transaction.
+ * Dry re-run on failure to obtain revert reason and call `options.showErrorNotification?`.
+ *
+ * @param  {Function} makeTx
+ * @param  {TxOptions} options?
+ * @returns Promise
+ */
+export async function tx(makeTx: Function, options?: TxOptions): Promise<void> {
```

Comment:
great work with this function!

Reply:
Awesome. Hoping it gets considered to be merged in https://github.com/Synthetixio/js-monorepo/tree/master/packages/transaction-notifier @clementbalestrat  ..

Reply:
We should probably catch up on this, to see if we want to use it everywhere in the dApp

Reply:
Awesome